test/integration/docker_save_test.js

```diff
@@ -0,0 +1,101 @@
+import assert from 'assert';
+import base from 'taskcluster-base'
+import Debug from 'debug';
+import Docker from 'dockerode-promise';
+import dockerOpts from 'dockerode-options';
+import DockerWorker from '../dockerworker';
+import fs from 'mz/fs';
+import https from 'https';
+import request from 'superagent-promise';
+import TestWorker from '../testworker';
+import zlib from 'zlib';
+
+let debug = Debug('docker-worker:test:docker-save-test');
+
+suite('use docker-save', () => {
+  let worker;
+  setup(async () => {
+    worker = new TestWorker(DockerWorker);
+    await worker.launch();
+  });
+
+  teardown(async () => {
+    if (worker) {
+      await worker.terminate();
+      worker = null;
+    }
+  });
+
+  test('run, then check contents', async () => {
+    let result = await worker.postToQueue({
+      payload: {
+        image: 'busybox',
+        command: ['/bin/sh', '-c', 'echo testString > /tmp/test.log'],
+        features: {
+          dockerSave: true
+        },
+        maxRunTime: 5 * 60
+      }
+    });
+
+    assert(result.run.state === 'completed', 'task should be successful');
+    assert(result.run.reasonResolved === 'completed',
+                 'task should be successful');
+
+    let taskId = result.taskId;
+    let runId = result.runId;
+
+    let signedUrl = worker.queue.buildSignedUrl(
```

Comment:
Not that it matters much here, but if you want to reduce a call to the queue, and since we no longer need a signed url, you could just build up the artifact url here without an external call.
```js
let url = `https://queue.taskcluster.net/v1/task/${taskId}/runs/${runId}/artifacts/public/dockerImage.tar`
```
            taskId + '/runs/' + runId + '/artifacts/' + path;